tweets/templatetags/tweet_tags.py

```diff
@@ -0,0 +1,15 @@
+from django import template
+
+register = template.Library()
+
+@register.filter
+def can_edit_tweet(user, tweet):
+  return user.id == tweet.user_id and tweet.original_tweet is None
+
+@register.filter
+def can_delete_tweet(user, tweet):
+  return user.id == tweet.user_id
+
+@register.filter
+def can_retweet(user, tweet):
+  return user.id != tweet.user_id and user.is_authenticated and tweet.original_tweet is None
```

Comment:
These don't look like tags

Reply:
means? there is nothing to be named here as tag. Although to filter data on html pages we call a term known as template tags